torch/csrc/utils.h

```diff
@@ -155,11 +155,16 @@
 #define THPUtils_assert(cond, ...) \
   THPUtils_assertRet(nullptr, cond, __VA_ARGS__)
 #define THPUtils_assertRet(value, cond, ...) \
-  if (THP_EXPECT(!(cond), 0)) {              \
-    THPUtils_setError(__VA_ARGS__);          \
+  if (C10_UNLIKELY_OR_CONST(!(cond))) {                               \
+    try { \
+      TORCH_CHECK((cond), ##__VA_ARGS__);  \
+    } catch (const std::exception& e) { \
+      THPUtils_setError(e.what());          \
```

Comment:
Most of the use above are already guarded with HANDLE_TH_ERRORS which will do this properly. No need to set the python error here.
For the few other cases, I think we can just add the HANDLE_TH_ERRORS macro where it is missing.

Reply:
OK, I see.